packages/auth/src/api/index.test.ts

```diff
@@ -365,6 +345,24 @@ describe('api/_performApiRequest', () => {
       expect(clock.clearTimeout).to.have.been.called;
       clock.restore();
     });
+
+    it('should handle network failure', async () => {
+      mockFetch.setUpWithOverride(() => {
+        return new Promise<never>((_, reject) =>
+          reject(new Error('network error'))
+        );
+      });
+      const promise = _performApiRequest<typeof request, never>(
+        auth,
+        HttpMethod.POST,
+        Endpoint.SIGN_UP,
+        request
+      );
+      await expect(promise).to.be.rejectedWith(
+        FirebaseError,
+        'auth/network-request-failed'
+      );
+    });
```

Comment:
nit: Can we check if "Network error" is propagated in error message?

Reply:
Done. This checks whether the [mock error message](https://github.com/firebase/firebase-js-sdk/pull/7125/files#diff-9c883d3a312965d9da48f7befccb9eb78c39ad990ef4384d3a081374eb3f6926R352) is propagated in error message.